### PR TITLE
Integration tests for confluent kakfa

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Added integration tests for Confluent kafka
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ X] New functionality includes testing.
- [X ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/kafka-plugins/README.md [hunk 1]
```diff
@@ -87,6 +87,86 @@ bin/kafka-server-start.sh config/server.properties
...
-
+./gradlew     data-prepper-plugins:kafka-plugins:integrationTest -Dtests.kafka.bootstrap_servers=<msk-bootstrap-servers> -Dtests.kafka.glue_registry_name=<glue-registry-name> -Dtests.kafka.glue_avro_schema_name=<glue-registry-avro-schema-name> -Dtests.kafka.glue_json_schema_name=<glue-registry-json-schema-name> -Dtests.msk.region=<msk-region> -Dtests.msk.arn=<msk-arn>  --tests "*TestAvroRecordConsumer*"
+
+```
+
+8. Command to run confluent integration tests (without schema registry)
+
+```
+./gradlew    data-prepper-plugins:kafka-plugins:integrationTest -Dtests.kafka.bootstrap_servers="pkc-lzvrd.us-west4.gcp.confluent.cloud:9092" -Dtests.kafka.topic_name=<topic-name> -Dtests.kafka.username=<confluent-cloud-api-key> -Dtests.kafka.password=<confluent-cloud-api-secret> --tests "*KafkaProducerConsumerIT*"
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/README.md:97]: Please replace `pkc-lzvrd.us-west4` with `<confluent-cloud-dns>`.

File: data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        String[] userInfoParts = schemaRegistryUserInfo.split(":");
+        when(schemaConfig.getSchemaRegistryApiKey()).thenReturn(userInfoParts[0]);
+        when(schemaConfig.getSchemaRegistryApiSecret()).thenReturn(userInfoParts[1]);
+        when(schemaConfig.getRegistryURL()).thenReturn(schemaRegistryUrl);
+    }
+
+    @Test
+    public void KafkaJsonProducerConsumerTest() {
+        when(schemaConfig.getVersion()).thenReturn(2);
+        Thread producerThread = new Thread(() -> {
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java:251]: As with my comment above - can we drop the threads? Typically tests should avoid using threads whenever possible.

File: data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+            producerThread.join();
+            consumerThread.join();
+        } catch (Exception e){}
+        assertThat(numRecordsReceived.get(), equalTo(numRecordsProduced));
+    }
+
+    public void consumeRecords(String servers, KafkaSourceConfig sourceConfig) {
+        kafkaSource = new KafkaSource(sourceConfig, pluginMetrics, acknowledgementSetManager, pipelineDescription);
+        kafkaSource.start(buffer);
+        for (int i = 0; i < 150; i++) {
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java:276]: Similar to my comments above. I think you don't need any of this. Use Awaitility.

File: data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+        } catch (SerializationException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    @Test
+    public void KafkaAvroProducerConsumerTest() {
+        when(schemaConfig.getVersion()).thenReturn(1);
+        Thread producerThread = new Thread(() -> {
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java:318]: Same as other thread comments.

File: data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+                avroRecord.put("message", testMessage+i);
+                avroRecord.put("ident", TEST_ID+i);
+                avroRecord.put("score", TEST_VALUE*(i+1));
+                String key = "key"+String.valueOf(i);
+                ProducerRecord<String, GenericRecord> record = new ProducerRecord<String, GenericRecord>(avroTopicName, key, avroRecord);
+                producer.send(record);
+            }
+            producer.flush();
+        } catch (SerializationException e) {
+            e.printStackTrace();
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java:372]: Same as above.

**Review by dlvenable — CHANGES_REQUESTED** [review]
At a high level, do we want to have integration tests which rely on Confluent resources? We need to get to the point where our integration tests run in GitHub, and this would require adding Confluent resources for those to run.

**Comment by kkondaka** [comment]
> At a high level, do we want to have integration tests which rely on Confluent resources? We need to get to the point where our integration tests run in GitHub, and this would require adding Confluent resources for those to run.

I do not think we can run these in GitHub but it is valuable to have this because we do not have a way to make sure Confluent Kafka works. Manually running these would give some idea on how to use Confluent Kafka with DataPrepper

**Review by dlvenable — DISMISSED** [review]
Nice, thanks! This is more readable in my view.

**Review by dlvenable — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Review by graytaylor0 — APPROVED** [review]

**Pull request merged**